canvas_modules/harness/cypress/integration/canvas/clipboard.js

```diff
@@ -21,6 +21,139 @@ describe("Test clipboard with no link selection enabled", function() {
 		cy.openCanvasDefinition("commentColorCanvas.json");
 	});
 
+	it("Test cut & paste in the right { x,y } positions when using keyboard", function() {
+		// Cut the node into the clipboard using keyboard (cmd+x)
+		cy.clickNode("DRUG1n");
+		cy.shortcutKeysCut();
+
+		// Move the mouse into {x,y} position & paste the node using (cmd+V) where the cursor is
+		cy.moveMouseToCoordinates(100, 100);
+		cy.shortcutKeysPaste();
+		cy.verifyNodeTransform("DRUG1n", 100, 100);
+
+		cy.clickNode("Na_to_K");
+		cy.shortcutKeysCut();
+
+		cy.moveMouseToCoordinates(500, 400);
+		cy.shortcutKeysPaste();
+		cy.verifyNodeTransform("Na_to_K", 618, 519);
+
+		cy.clickNode("C5.0");
+		cy.shortcutKeysCut();
+
+		cy.moveMouseToCoordinates(300, 400);
+		cy.shortcutKeysPaste();
+		cy.verifyNodeTransform("C5.0", 411, 400);
+	});
+
+	it("Test cut & paste in default {x,y} positions when mousing outside the canvas using keyboard cmd+v", function() {
+		// Cut the node into the clipboard
+		cy.clickNode("DRUG1n");
+		cy.shortcutKeysCut();
+
+		// Paste the node into default {x,y} positions
+		cy.moveMouseToCoordinates(0, 100);
+		cy.shortcutKeysPaste();
+		cy.verifyNodeTransform("DRUG1n", 96, 219);
+
+		cy.clickNode("Na_to_K");
+		cy.shortcutKeysCut();
+
+		cy.moveMouseToCoordinates(0, 100);
+		cy.shortcutKeysPaste();
+		cy.verifyNodeTransform("Na_to_K", 218, 219);
+	});
+
+	it("Test copy & paste in the right { x,y } positions  when using keyboard", function() {
+		// Copy the node into the clipboard
+		cy.clickNode("DRUG1n");
+		cy.shortcutKeysCopy();
+
+		// Make DRUG1n node editable & change it into DRUG2n
```

Comment:
This is an example of what I was talking about the other day where, in comments,  the “why” is more interesting than the “what”.
// Make DRUG1n node editable & change it into DRUG2n
cy.setCanvasConfig({ "selectedNodeLayout": { labelEditable: true, labelWidth: 200 } });
cy.hoverOverNodeLabel("DRUG1n");
cy.clickNodeLabelEditIcon("DRUG1n");
cy.enterLabelForNode("DRUG1n", "DRUG2n");
You have documented “what” the code is doing. However, what you missed out is the “why” — which is much more interesting to the developer reading your comment.  In other words, “why” is the test renaming a node here.  The reason, of course, is that we want to maintain unique node labels on the canvas for the test. So the comment could have said :
// Make DRUG1n node editable & change it into DRUG2n to make sure we have
// unique node labels on the canvas